From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 7 Nov 2017 00:01:04 -0500
Subject: [PATCH] EMC: Don't use snapshots for TileEntity::getOwner

Original license: MIT
Original project: https://github.com/starlis/empirecraft

Also see Leaf's EMC-Don-t-use-snapshots-for-acquiring-blockstate

diff --git a/net/minecraft/world/level/block/entity/BlockEntity.java b/net/minecraft/world/level/block/entity/BlockEntity.java
index 7cbb92c23f0b3d49b8ea6cd66ed4f377dd22ee98..0527e14039a928605c485828e69c29a913032517 100644
--- a/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -386,7 +386,7 @@ public abstract class BlockEntity implements DebugValueSource {
 
     // CraftBukkit start - add method
     public org.bukkit.inventory.@Nullable InventoryHolder getOwner() {
-        return getOwner(true);
+        return getOwner(org.dreeam.leaf.config.modules.opt.TileEntitySnapshotCreation.enabled); // Leaf - EMC - Don't use snapshots for TileEntity::getOwner
     }
 
     public org.bukkit.inventory.@Nullable InventoryHolder getOwner(boolean useSnapshot) {
